lib/menubar/icon_theme.lua

```diff
@@ -0,0 +1,262 @@
+---------------------------------------------------------------------------
+--- Class module for icon lookup for menubar
+--
+-- @author Kazunobu Kuriyama
+-- @copyright 2015 Kazunobu Kuriyama
+-- @release @AWESOME_VERSION@
+-- @classmod menubar.icon_theme
+---------------------------------------------------------------------------
+
+-- This implementation is based on the specifications:
+--  Icon Theme Specification 0.12
+--  http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-0.12.html
+
+local beautiful = require("beautiful")
+local awful = require("awful")
+local GLib = require("lgi").GLib
+local Gio = require("lgi").Gio
+local index_theme = require("menubar.index_theme")
+
+local ipairs = ipairs
+local setmetatable = setmetatable
+local string = string
+local table = table
+local math = math
+local io = io
+
+-- Returns a table whose element is a path used for icon lookup.
+-- The names of the directories and the order of them are based on the spec.
+local get_default_base_directories = function()
+    local dirs = {}
+
+    table.insert(dirs, GLib.get_home_dir() .. "/.icons")
+    for _, dir in ipairs(GLib.get_system_data_dirs()) do
+        table.insert(dirs, dir .. "/icons")
+    end
+    table.insert(dirs, "/usr/share/pixmaps")
+
+    return dirs
+end
+
+local is_readable_directory = function(path)
```

Comment:
Please move this to `lib/awful/util.lua`.
And then `util.file_readable` could make use of it, too - via a common method.